motionSetPointsApp/src/setPoint.cpp

```diff
@@ -31,83 +60,105 @@ struct CaselessCompare {
 	}
 };
 
-/* Load the lookup file if it is not already loaded */
-void checkLoadFile(const char* env_fname) {
-	std::string key(env_fname);
-	if ( gTables.count(key)==0 ) {
-		loadDefFile(env_fname);
-	}
-}
-
-/* Load the lookup file named by the environment variable */
-void loadDefFile(const char* env_fname) {
+/* Load the lookup file named by the environment variable.
+   Arguments:
+     const char *env_fname              [in] The environment variable that contains the path to the file
+            int  expectedNumberOfCoords [in] The number of coordinates expected in the file
+ */
+void loadDefFile(const char* env_fname, int expectedNumberOfCoords) {
+    FileIO fileIO;
 	const char *fname = getenv(env_fname);
 	if ( fname==NULL ) {
 		errlogSevPrintf(errlogMajor, "motionSetPoints: Environment variable \"%s\" not set\n", env_fname);
 		return;
 	}
-	loadFile(fname, env_fname);
+	loadFile(&fileIO, fname, env_fname, expectedNumberOfCoords);
 }
 
-// Find the table structure for a key
+/* Get the table structure for a key.
+   Arguments:
+     const char *env_fname [in] Key to identify file
+   Return:
+     A reference to the lookup table
+ */
 LookupTable& getTable(const char *env_fname) {
 	std::string key(env_fname);
     epicsGuard<epicsMutex> _lock(g_lock); // need to protect map as this may create entry	
 	LookupTable& table = gTables[key];
 	return table;
 }
 
-// Load a lookup file
-// Arguments
-//   const char *fname     [in] File to load
-//   const char *env_fname [in] Key to identify file
-void loadFile(const char *fname, const char *env_fname) {
+/* Create a table row from a line in the lookup file.
+   Arguments:
+     string fileLine [in] Line from file
+   Return:
+     The created row
+ */
+LookupRow createRowFromFileLine(std::string fileLine) {
+    LookupRow row;
+    std::stringstream ss(fileLine);
+    std::istream_iterator<std::string> begin(ss);
+    std::istream_iterator<std::string> end;
+    std::vector<std::string> vstrings(begin, end);
+    std::strcpy(row.name, vstrings[0].c_str());
+    std::transform(std::next(vstrings.begin()), vstrings.end(), back_inserter(row.coordinates),
+        [](std::string const& val) {return std::stod(val); });
+    return row;
+}
+
+/* Load a lookup file
+   Arguments:
+     FileIOInterface *fileIO                  [in] Interface to interact with the file system
+     const char      *fname                   [in] File to load
+     const char      *env_fname               [in] Key to identify file
+     int              expectedNumberOfCoords  [in] Expected number of coordinates
+ */
+void loadFile(FileIOInterface *fileIO, const char *fname, const char *env_fname, int expectedNumberOfCoords) {
```

Comment:
I'm not particularly a fan of logging an error and not displaying any error to the user only an empty table. I feel it would be better to return some kind of error if we cannot load the file.

Reply:
As discussed, ideally this error would propagate to an EPICS alarm but given it's only likely to happen on first start of the IOC and it should be quite obvious to the user as there are no available positions I have put it in a new ticket https://github.com/ISISComputingGroup/IBEX/issues/5766